docs/conceptual/core/executor.md

```diff
@@ -2,8 +2,147 @@
 title: Executor
 ---
 
-* Runtime Versioning
+The executor is responsible for dispatching and executing calls into the
+Substrate runtime.
 
-* Native VS Wasm Execution
+## Runtime Execution
 
-* Forkless Runtime Upgrades
\ No newline at end of file
+The Substrate runtime is compiled into a native executable and a WebAssembly
+(Wasm) binary.
+
+The native runtime is included as part of the node executable, while the Wasm
+binary is stored on the blockchain under a well known storage key.
+
+These two representations of the runtime may not be the same. For example: when
+the runtime is upgraded. The executor determines which version of the runtime to
+use when dispatching and executing calls.
+
+### Execution Strategy
+
+Before runtime execution begins, the Substrate client chooses which runtime
+execution environment should be used. This is controlled by the execution
+strategy, which can be configured for the different parts of the blockchain
+execution process. The strategies are:
+
+- `Native`: Execute with native build (if available, WebAssembly otherwise).
+- `Wasm`: Only execute with the WebAssembly build.
+- `Both`: Execute with both native (where available) and WebAssembly builds.
+- `NativeElseWasm`: Execute with the native build if possible; if it fails, then
+  execute with WebAssembly.
+
+The default execution strategies for the different parts of the blockchain
+execution process are:
+
+- Syncing: `NativeElseWasm`
+- Block Import: `NativeElseWasm`
+- Block Construction: `Wasm`
+- Off-Chain Worker: `Native`
+- Other: `Native`
+
+### Wasm Execution
+
+The Wasm representation of the Substrate runtime is considered the canonical
+runtime. Because this Wasm runtime is placed in the blockchain storage, the
+network must come to consensus about this binary. Thus it can be verified to be
+consistent across all syncing nodes.
+
+The Wasm execution environment can be more restrictive than the native execution
+environment. For example, the Wasm runtime always executes in a 32-bit
+environment with a 4 GB memory limit.
+
+For these reasons, the blockchain prefers to do block construction with the Wasm
```

Comment:
This sort of implies: 

> for the reason of wasm being much more limited 

 Then 

> the blockchain prefers to do block construction with the Wasm

While I suppose you are referring to the fact that wasm runtime is canonical and hence it must be used as the production reference, despite being slower.

Reply:
I don't think I mentioned anything about "slow" here.

I am simply saying that Wasm is more restrictive, as in something executed in Wasm will always work in Native, but not vice-versa.

Hence, to create fully compatible blocks with all nodes, you must use Wasm.